### PR TITLE
[ENG-8721][docs] document new custom build functions

### PR DESCRIPTION
**Pull request by dsokal**
Documents two new built-in functions for custom builds: `eas/use_npm_token` and `eas/install_node_modules`

### CODE REVIEW

**Review by szdziedzic — APPROVED** [review]

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 1]
```diff
@@ -540,6 +540,39 @@ build:
         run: ls assets
 ```
 
+#### `eas/use_npm_token`
+
+Configures node package managers (npm / yarn / pnpm) for use with private packages, published either to npm or private registry.
```
Comment (amandeepmittal) [inline docs/pages/preview/custom-build-config-schema.mdx:545]:
```suggestion
Configures node package managers (npm, pnpm, or Yarn) to use with private packages, published either to npm or private registry.
```

[External product casing](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#follow-external-product-casing)

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 2]
```diff
@@ -540,6 +540,39 @@ build:
...
+Set `NPM_TOKEN` in your projects secrets and this function will configure the build environment by creating `.npmrc` with the token.
```
Comment (amandeepmittal) [inline docs/pages/preview/custom-build-config-schema.mdx:546]:
```suggestion
Set `NPM_TOKEN` in your project's secrets, and this function will configure the build environment by creating **.npmrc** with the token.
```

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 3]
```diff
@@ -540,6 +540,39 @@ build:
...
+    - eas/use_npm_token
+    # @end #
+    - run:
+        name: Install dependencies
+        run: npm install # <---- Can now install private packages
+```
+
+#### `eas/install_node_modules`
+
+Installs node modules using the package manager detected base on your project. Works with monorepos.
```
Comment (amandeepmittal) [inline docs/pages/preview/custom-build-config-schema.mdx:563]:
```suggestion
Installs node modules using the package manager (npm, pnpm, or Yarn) detected based on your project. Works with monorepos.
```

File: docs/pages/preview/custom-build-config-schema.mdx [hunk 4]
```diff
@@ -540,6 +540,39 @@ build:
...
+Supported package managers: npm, yarn, pnpm
```
Comment (amandeepmittal) [inline docs/pages/preview/custom-build-config-schema.mdx:564]:
```suggestion
```

I think we can list the package managers above instead of here.

**Pull request merged**